### ISSUE TITLE
Calculate difficulty for a simfile and show it as a separate value from set difficulty

### ISSUE THREAD

**Issue by ArcFutahito** [stepmania/stepmania#1293]
It's a little bit hard to understand difficulty standards for a newbie and simfile creators don't really follow some unified standard for setting simfile difficulty. I think it would be great to display a weighted difficulty value so everyone can see the actual calculated difficulty.
How to calculate: 
Break up simfile in 30 second chunks
Calculate steps per second for each chunk
Calculate average bpm deviation for a chunk
See if spikes are present in any chunk of simfile and add some extra points if those exist
So, generally that would look something like this: 
1) 10 steps per second + 0 bpm deviation = normal chunk
2) 20 steps per second + 50 bpm deviation = hard chunk
3) 40 steps per second + 0 bpm deviation = expert chunk
4) 35 steps per second + 10 bpm deviation = expert chunk
A total length of simfile 1:30 to 2:00
Total difficulty: expert

By doing this thing you can also calculate all sorts of stuff to help players evaluate their performance more accurately and not get confused by difficulty 1 dumps.

**Comment by SpoOkyMagician** [comment]
Yeah, a(n) auto difficulty foot/meter would be nice. While making your own opinion number foot/meter is nice, there is no standard and can really throw off players. The only way to know a steps difficulty right now is to play it. (Maybe read the description and get an idea. However, it can just be an outright lie/joke.) This would be a nice feature.

**Comment by JapanYoshi** [comment]
CyberiaStyle has an option to calculate the CyberiaStyle original difficulty scale. It's better to use the theme instead.
